simple/reprovide.go

```diff
@@ -216,7 +216,7 @@ func pinSet(ctx context.Context, pinning Pinner, dag ipld.DAGService, onlyRoots
 			if onlyRoots {
 				set.Visitor(ctx)(key)
 			} else {
-				err := merkledag.Walk(ctx, merkledag.GetLinksWithDAG(dag), key, set.Visitor(ctx))
+				err := merkledag.Walk(ctx, merkledag.GetLinksWithDAG(dag), key, set.Visitor(ctx), merkledag.Concurrent(), merkledag.WithRoot())
```

Comment:
So, I missed that when reviewing the PR. I've reverted the default root behavior in https://github.com/ipfs/go-merkledag/pull/43 to avoid silently changing the API. Released in 0.2.2.